File: 538. Convert BST to Greater Tree/solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var convertBST = function(root) {
  let sum = 0;
  const helper = node => {
    if (!node) return;
    helper(node.right);
    node.val += sum;
    sum = node.val;
    helper(node.left);
  };
  helper(root);
  return root;
};

File: 538. Convert BST to Greater Tree/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def convertBST(self, root: TreeNode) -> TreeNode:
        def dfs(root, sum):
            if not root:
                return sum

            root.val += dfs(root.right, sum)
            sum = root.val
            return dfs(root.left, sum)

        dfs(root, 0)
        return root
